tests/e2e/testdata/scenarios/direct/llm_custom_diff/value_type/label/script.yaml

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: logging.cnrm.cloud.google.com/v1beta1
+kind: LoggingLogMetric
+metadata:
+  name: logginglogmetric-${uniqueId}
+  annotations:
+    cnrm.cloud.google.com/state-into-spec: absent
+spec:
+  filter: "resource.type=gae_app AND severity<=ERROR"
+  metricDescriptor:
+    displayName: "a concise name"
+    labels:
+    - description: "a label description"
+      key: "testkey"
+      valueType: "STRING"
+    metricKind: "DELTA"
+    valueType: "DISTRIBUTION"
+  valueExtractor: "EXTRACT(jsonPayload.response)"
+  labelExtractors:
+    testkey: "EXTRACT(jsonPayload.request)"
+  bucketOptions:
+    explicitBuckets:
+      bounds:
+      - 1.5
+      - 4.5
+  projectRef:
+    external: "projects/${projectId}"
+---
+apiVersion: logging.cnrm.cloud.google.com/v1beta1
+kind: LoggingLogMetric
+metadata:
+  name: logginglogmetric-${uniqueId}
+  annotations:
+    cnrm.cloud.google.com/state-into-spec: absent
+spec:
+  filter: "resource.type=gae_app AND severity<=ERROR"
+  metricDescriptor:
+    displayName: "a concise name"
+    labels:
+    - description: "a label description"
+      key: "testkey"
+      valueType: ""
```

Comment:
This may be a bit outside of the scope of custom diff: What's the POST request when this value is `""` in the direct controller? In the DCL controller, field seems to be dropped by the request:

`{"bucketOptions":{"explicitBuckets":{"bounds":[1.5,4.5]}},"filter":"resource.type=gae_app AND severity\u003c=ERROR","labelExtractors":{"testkey":"EXTRACT(jsonPayload.request)"},"metricDescriptor":{"displayName":"a concise name","labels":[{"description":"a label description","key":"testkey"}],"metricKind":"DELTA","valueType":"DISTRIBUTION"},"name":"logginglogmetric-e3leul22li5i2y55gzaa","valueExtractor":"EXTRACT(jsonPayload.response)"}`

For compatibility, I think most field (if not all) of empty values will be dropped during the POST. Just want to double check - is it how direct controllers handle resource creations?

Reply:
> For compatibility, I think most field (if not all) of empty values will be dropped during the POST

I think we can drop it as a parity at first then decide if we want to emulate that.

Reply:
Just to double check - is "dropping the field of empty values" an existing behavior in the direct controller, or is it something we plan to do?